gitlint/tests/test_git.py

```diff
@@ -84,45 +80,42 @@ def git_log_side_effect(*args, **_kwargs):
                                                              tzinfo=dateutil.tz.tzoffset("+0100", 3600)))
         self.assertListEqual(last_commit.parents, [u"åbc", "def"])
         self.assertTrue(last_commit.is_merge_commit)
-
-        # assert that changed files are read using git command
-        sh.git.assert_called_once_with('diff-tree', '--no-commit-id', '--name-only', '-r', 'HEAD',
-                                       **expected_sh_special_args)
         self.assertListEqual(last_commit.changed_files, ["file1.txt", u"påth/to/file2.txt"])
 
     @patch('gitlint.git.sh')
     def test_get_latest_commit_command_not_found(self, sh):
-        sh.git.log.side_effect = CommandNotFound("git")
+        sh.git.side_effect = CommandNotFound("git")
         expected_msg = "'git' command not found. You need to install git to use gitlint on a local repository. " + \
                        "See https://git-scm.com/book/en/v2/Getting-Started-Installing-Git on how to install git."
         with self.assertRaisesRegex(GitContextError, expected_msg):
             GitContext.from_local_repository(u"fåke/path")
 
         # assert that commit message was read using git command
-        sh.git.log.assert_called_once_with('-1', '--pretty=%B', _tty_out=False, _cwd=u"fåke/path")
+        sh.git.assert_called_once_with("rev-list", "HEAD^...", _tty_out=False, _cwd=u"fåke/path")
 
     @patch('gitlint.git.sh')
     def test_get_latest_commit_git_error(self, sh):
         # Current directory not a git repo
         err = b"fatal: Not a git repository (or any of the parent directories): .git"
-        sh.git.log.side_effect = ErrorReturnCode("git log -1 --pretty=%B", b"", err)
+        sh.git.side_effect = ErrorReturnCode("git rev-list HEAD^...", b"", err)
 
         with self.assertRaisesRegex(GitContextError, u"fåke/path is not a git repository."):
             GitContext.from_local_repository(u"fåke/path")
 
         # assert that commit message was read using git command
-        sh.git.log.assert_called_once_with('-1', '--pretty=%B', _tty_out=False, _cwd=u"fåke/path")
-
-        sh.git.log.reset_mock()
+        sh.git.assert_called_once_with("rev-list", "HEAD^...",
+                                       _tty_out=False, _cwd=u"fåke/path")
+        sh.git.reset_mock()
         err = b"fatal: Random git error"
-        sh.git.log.side_effect = ErrorReturnCode("git log -1 --pretty=%B", b"", err)
+        sh.git.side_effect = ErrorReturnCode("git rev-list HEAD^...", b"", err)
 
-        expected_msg = u"An error occurred while executing 'git log -1 --pretty=%B': {0}".format(err)
+        expected_msg = u"An error occurred while executing 'git rev-list HEAD^...': {0}".format(err)
```

Comment:
You need to escape ```^``` in ```expected_msg``` because the check below is a regex check, not an exact match check. So ```expected_msg``` should be:
```python
expected_msg = u"An error occurred while executing 'git rev-list HEAD\^...': {0}".format(err)
```